### PR TITLE
Implementation of the database queries

### PR DESCRIPTION
**Pull request by nicolasp9-dev**
This branch implements some new classes that are useful to perform elements request to the database. You can get back some list of one kind of element. Some changes have also be done to the favorFragment to allow a more dynamic (texts) and encapsulated interface to edit or create a new favour with the same fragment.

### CODE REVIEW

**Comment by nicolasp9-dev** [comment]
@alexander-san , I appreciate a lot your capacity to comment others work and show that you're here, especially during meetings. We are doing a development project, which means that we are coding many stuff and reporting it in the most convenient way, I think I'm providing many details on what I do, in commits or in slack. You're the one complaining about a lack of informations on database but during lasts week we met a lot with others to talk about important subject like the database. You're the only one that never showed up, never sent me a private message or never call me, but you dared to blame me that you could not work because database API wasn't defined and ready. I can't accept to work like that, we are a team we need to stay aware of what others do as if it's not in detail. I don't want to work with someone that only cares about GitHub and meeting appearances, and it'll be a pleasure to take an hour with you to explain you what have been done and you'll be able to do reviews without more needs of informations.

**Comment by Jmion** [comment]
I think that providing some basic doc, at least at the class level of what each thing is supposed to do would be a good idea. Since @AndreaScalisi @Charrlyine @jeniffer9 @alexander-san are not that familiar with it. It is also to ensure that everybody uses the amazing work that you have done in the appropriate way. It would be sad to see some people have trouble understanding how to use this. I think that at least a top level Javadoc description of what each class can do is a good thing. For the method documentation maybe providing a bit for the methods that are going to be called from other parts of the app might be good. 
Other than the documentation and a few minor little things that I saw on the way through I think that you did a very good job. Thanks

File: app/src/main/java/ch/epfl/sweng/favors/ProfileFragment.java [hunk 1]
```diff
@@ -17,11 +17,13 @@
...
-    public ObservableField<String> baseCity = User.getMain().getObservableStringObject(User.StringFields.basedLocation);
-    public ObservableField<String> sex = User.UserGender.getObservableGenderString(User.getMain());//User.getMain().getObservableStringObject(User.StringFields.sex);
-    public ObservableField<String> email = User.getMain().getObservableStringObject(User.StringFields.email);
+
+    public ObservableField<String> firstName = User.getMain().getObservableObject(User.StringFields.firstName);
+    public ObservableField<String> lastName = User.getMain().getObservableObject(User.StringFields.lastName);
+    public ObservableField<String> baseCity = User.getMain().getObservableObject(User.StringFields.location);
+    public ObservableField<String> sex = User.getMain().getObservableObject(User.StringFields.sex);
+    public ObservableField<String> email = User.getMain().getObservableObject(User.StringFields.email);
+
```
[the remark this answers is not in the document]
Comment (Jmion) [reply app/src/main/java/ch/epfl/sweng/favors/ProfileFragment.java]: You overloaded the get method?

[the remark this answers is not in the document]
Comment (nicolasp9-dev) [reply app/src/main/java/ch/epfl/sweng/favors/ProfileFragment.java]: Yes, we now can get every kind of observable fields from the getObservableObjectMethod, which mean that you can observe an object (like Location / Coordinates) or a Boolean

File: app/src/main/java/ch/epfl/sweng/favors/SetUserInfo.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public void afterTextChanged(Editable editable) {
     private TextWatcherCustom basedLocationWatcher = new TextWatcherCustom() {
         @Override
         public void afterTextChanged(Editable editable) {
-            User.getMain().set(User.StringFields.basedLocation, editable.toString());
+            User.getMain().set(User.StringFields.location, editable.toString());
```
[the remark this answers is not in the document]
Comment (Jmion) [reply app/src/main/java/ch/epfl/sweng/favors/SetUserInfo.java]: yes

**Pull request merged**